skyportal/handlers/api/reminder.py

```diff
@@ -0,0 +1,900 @@
+                    shift_id, self.current_user, raise_if_none=True
+                )
+            except AccessError:
+                return self.error(f'Could not access Shift {shift.id}.', status=403)
+            for user in users:
+                reminder = ReminderOnShift(
+                    text=reminder_text,
+                    shift_id=shift.id,
+                    author=author,
+                    groups=groups,
+                    bot=is_bot_reminder,
+                    next_reminder=next_reminder,
+                    reminder_delay=reminder_delay,
+                    number_of_reminders=number_of_reminders,
+                    users=users,
+                )
+        else:
+            return self.error(f'Unknown resource type "{associated_resource_type}".')
+
+        if associated_resource_type.lower() == "sources":
+            text_to_send = f"*@{self.associated_user_object.username}* created a reminder on *{obj_id}*"
+            url_endpoint = f"/source/{obj_id}"
+        elif associated_resource_type.lower() == "spectra":
+            text_to_send = f"*@{self.associated_user_object.username}* created a reminder on *{spectrum_id}*"
+            url_endpoint = f"/source/{spectrum_id}"
+        elif associated_resource_type.lower() == "gcn_event":
+            text_to_send = f"*@{self.associated_user_object.username}* created a reminder on *{gcnevent_id}*"
+            url_endpoint = f"/gcn_events/{gcnevent_id}"
+        elif associated_resource_type.lower() == "shift":
+            text_to_send = f"*@{self.associated_user_object.username}* created a reminder on *shift {shift_id}*"
+            url_endpoint = "/shifts"
+        else:
+            return self.error(f'Unknown resource type "{associated_resource_type}".')
+
+        for user in users:
+            DBSession().add(
+                UserNotification(
+                    user=user,
+                    text=text_to_send,
+                    notification_type="mention",
+                    url=url_endpoint,
+                )
+            )
+
+        DBSession().add(reminder)
+        self.verify_and_commit()
+        for user in users:
+            self.flow.push(user.id, "skyportal/FETCH_NOTIFICATIONS", {})
+
+        if hasattr(reminder, 'obj'):  # reminder on object, or object related resources
+            self.push_all(
+                action='skyportal/REFRESH_SOURCE',
+                payload={'obj_key': reminder.obj.internal_key},
+            )
+
+        if isinstance(reminder, ReminderOnGCN):
+            self.push_all(
+                action='skyportal/REFRESH_GCNEVENT',
+                payload={'gcnEvent_dateobs': reminder.gcn.dateobs},
+            )
+        elif isinstance(reminder, ReminderOnSpectrum):
+            self.push_all(
+                action='skyportal/REFRESH_SOURCE_SPECTRA',
+                payload={'obj_internal_key': reminder.obj.internal_key},
+            )
+        elif isinstance(reminder, ReminderOnShift):
```

Comment:
I'm saying this in the comment above because I am a bit worried about that. Refreshing the shifts everytime a reminder is added on a shift seems like it will happen all the time no?

Reply:
yeah removed.